test/e2e/canary_test.go

```diff
@@ -142,3 +147,75 @@ func buildCanaryCurlPod(name, namespace, image, host string) *corev1.Pod {
 		},
 	}
 }
+
+// TestCanaryRouteRotationAnnotation verifies that the
+// canary controller respect the canary route rotation
+// annotation for the default ingress controller.
+func TestCanaryRouteRotationAnnotation(t *testing.T) {
+	// Set the CanaryRouteRotation annotation to true
+	// on the default ingress controller.
+	if err := setDefaultIngressControllerRotationAnnotation(t, true); err != nil {
+		t.Fatalf("failed to set canary route rotation annotation: %v", err)
+	}
+
+	defer setDefaultIngressControllerRotationAnnotation(t, false)
```

Comment:
How does the test fail if this fails?

Reply:
fixed so the error is caught and we fail on that need be.